packages/mock-block-dock/src/debug-view/dev-tools/datastore-graph-visualisation.tsx

```diff
@@ -0,0 +1,196 @@
+import { Entity, Link } from "@blockprotocol/graph/.";
+import { GraphChart, GraphSeriesOption } from "echarts/charts";
+import * as echarts from "echarts/core";
+import { SVGRenderer } from "echarts/renderers";
+import { useEffect, useRef, useState } from "react";
+
+import { useMockBlockDockContext } from "../../mock-block-dock-context";
+
+const parseNameFromEntity = (entity: Entity): string =>
+  typeof entity.properties.name === "string"
+    ? entity.properties.name
+    : entity.entityId;
+
+const graphHeight = 500;
+
+type SeriesOption = GraphSeriesOption;
+
+// Combine an Option type with only required components and charts via ComposeOption
+type EChartOption = echarts.ComposeOption<SeriesOption>;
+
+const createDefaultEChartOptions = (params?: {
+  initialNodes?: GraphSeriesOption["nodes"];
+  initialEdges?: GraphSeriesOption["edges"];
+}): EChartOption => ({
+  color: ["#6F59EB"],
+  series: [
+    {
+      type: "graph",
+      layout: "force",
+      draggable: true,
+      /** @todo: find way to get this working, currently interferes with `draggable` */
+      // roam: true,
```

Comment:
@CiaranMn we have a choice of either:
1. `draggable: false` and `roam: true` (user cannot rearrange nodes, but can pan the visualisation) or
2. `draggable: true` and `roam: false` (user can rearrange nodes, but can't pan the visualisation).

I've opted for 2 for now. This works fine for small graphs, but will be problematic for graphs that exceed the size of the wrapping container (either because they're very large, or because there is a small screen width). Let me know if you'd like me to change this.

Reply:
I think we're better off with option (1) – I think it'll be more important for users to be able to explore the graph than to rearrange it. Let me know if there's an advantage to rearranging that would change this (is it hard to see without rearranging in some circumstances?)

Reply:
> is it hard to see without rearranging in some circumstances?

Sometimes yes, when labels overlap for instance.

Reply:
I don't think so, based on [their docs](https://apache.github.io/echarts-handbook/en/concepts/chart-size/) and examples it seems you have to tell echarts the dimensions of the container before it even starts creating the graph. I haven't come across some way of extracting how big the produced graph is if it exceeds the bounds of the container.

It might be possible to give it a temporary dimension, let it produce the graph and then manually query the SVG in the HTML tree to figure out how big the graph is and then readjust the container - but I think that would be overkill

Reply:
Fair enough - let's leave it how it is for now